ladybug/datacollection.py

```diff
@@ -524,6 +524,24 @@ def average_monthly_for_each_hour(self):
         """
         return self.average_data_monthly_for_each_hour(self.values)
 
+    def data_to_HOY(self, condition="x>5"):
+        """
+        Provides a list of HOY for the datapoints in the
+        filtered data from a conditional statement.
+        """
+        if isinstance(condition, str):
+            pass
```

Comment:
If this is for type checking then you can be explicit about it. The check and pass is confusing. I suggest to do:

```
if not isinstance(condition, str):
   # rest of the code
```

or just use assert

```python
assert isinstance(condition, str), ValueError(error_message)
```

Reply:
Just to share something I learned recently. In Python, the use of assertions shall be limited to the conditions in code that are not easily covered by the exceptions. With -o and -oo command line switches, the assert statements can be turned into null operations. Basis are [this](https://docs.python.org/3/using/cmdline.html#cmdoption-o) and [that](https://stackoverflow.com/questions/944592/best-practice-for-python-assert)